### ISSUE TITLE
Complete documentation and/or firmware source code

### ISSUE THREAD

**Issue by DenisD3D** [Macrobase-tech/Emm42#3]
Hi,
Is it possible to get access to the Emm42 complete documentation (containing all available serial command).
We would also love to have access to the firmware source code to see all available features.
Thanks in advance
Denis

**Comment by Macrobase-tech** [comment]
I am sorry that supply doesn't provide code source.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Macrobase-tech/Emm42"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年1月12日(星期四) 晚上8:45
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[Macrobase-tech/Emm42] Complete documentation and/or firmware source code (Issue #3)





 
Hi,
 Is it possible to get access to the Emm42 complete documentation (containing all available serial command).
 We would also love to have access to the firmware source code to see all available features.
 Thanks in advance
 Denis

**Comment by DenisD3D** [comment]
Is it possible to at least get access to complete documentation (containing serial commands, ...) ?

**Comment by Macrobase-tech** [comment]
I updated some files,can it help?




1005918473
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Macrobase-tech/Emm42"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年1月13日(星期五) 晚上7:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Macrobase-tech/Emm42] Complete documentation and/or firmware source code (Issue #3)





 
Is it possible to at least get access to complete documentation (containing serial commands, ...) ?

**Comment by DenisD3D** [comment]
Yes it's perfect (although an English version would be even better :) )

**Comment by Macrobase-tech** [comment]
Great.
Files are provided from supplier.
I will update EN ASAP.




1005918473
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Macrobase-tech/Emm42"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年1月14日(星期六) 晚上11:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Macrobase-tech/Emm42] Complete documentation and/or firmware source code (Issue #3)





 
Yes it's perfect (although an English version would be perfect :) )

**Comment by Nathan22211** [comment]
I second the English translation of the datasheet. At the very least, run it through Baidu translate or something similar. I've been writing scripts to control and monitor closed-loop steppers from Klipper, and these are my next target, so I need the serial commands and what they do

**Comment by Macrobase-tech** [comment]
Excellent.
serial commands already upload to Github.
pulse,RS232,RS485,Serial port







1005918473
***@***.***



&nbsp;




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "Macrobase-tech/Emm42"                                                                                    ***@***.***&gt;;
Date:&nbsp;Fri, Feb 17, 2023 05:18 AM
***@***.***&gt;;
***@***.******@***.***&gt;;
Subject:&nbsp;Re: [Macrobase-tech/Emm42] Complete documentation and/or firmware source code (Issue #3)





 
I second the English translation of the datasheet. At the very least, run it through Baidu translate or something similar. I've been writing scripts to control and monitor closed-loop steppers from Klipper, and these are my next target, so I need the serial commands and what they do

**Comment by Nathan22211** [comment]
It could just be VS code, but with the comments in the source code being in Chinese, most characters in the comments don't come up as characters that would be readable even if I understood mandarin. I might see if google translate picks them up but there's no guarantee there

**Comment by Nathan22211** [comment]
This is what I got after google translating the comments in the audrino.ino

> Install the module according to the Emm42_V4.x manual 
> * Click Cal to calibrate the encoder at the first power-on
> * After the calibration is completed, select the CR_UART serial port control mode in the Mode option
> * The baud rate of the default module is 38400, and the address is 1

> Speed gear 0xE8, acceleration 0, pulse number 0x0C80 = 3200 pulses

not sure what to make of that but I think I might have figured out how to translate the datasheet. Do you want me to PR the translated one?

**Comment by Macrobase-tech** [comment]
The speed gear and acceleration gear are only a gear. The actual speed and acceleration will vary according to the actual load. Good to measure actual speed.
1005918473
***@***.***



&nbsp;




------------------&nbsp;Original&nbsp;------------------
From:                                                                                                                        "Macrobase-tech/Emm42"                                                                                    ***@***.***&gt;;
Date:&nbsp;Sun, Feb 19, 2023 04:19 AM
***@***.***&gt;;
***@***.******@***.***&gt;;
Subject:&nbsp;Re: [Macrobase-tech/Emm42] Complete documentation and/or firmware source code (Issue #3)





 
This is what I got after google translating the comments in the audrino.ino
  
Install the module according to the Emm42_V4.x manual
  
Click Cal to calibrate the encoder at the first power-on
 
After the calibration is completed, select the CR_UART serial port control mode in the Mode option
 
The baud rate of the default module is 38400, and the address is 1
    
Speed gear 0xE8, acceleration 0, pulse number 0x0C80 = 3200 pulses
  
not sure what to make of that but I think I might have figured out how to translate the datasheet. Do you want me to PR the translated one?
